test/e2e/control_plane_safe_eviction_test.go

```diff
@@ -0,0 +1,73 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
+	cpoetcd "github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/etcd"
+	hcpmanifests "github.com/openshift/hypershift/hypershift-operator/controllers/manifests"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	corev1 "k8s.io/api/core/v1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestControlPlaneSafeEviction(t *testing.T) {
+	t.Parallel()
+	g := NewWithT(t)
+
+	client, err := e2eutil.GetClient()
+	g.Expect(err).NotTo(HaveOccurred(), "failed to get k8s client")
+
+	ctx, cancel := context.WithCancel(testContext)
+	defer cancel()
+
+	clusterOpts := globalOpts.DefaultClusterOptions(t)
+
+	hostedCluster := e2eutil.CreateCluster(t, ctx, client, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir, globalOpts.ServiceAccountSigningKey)
```

Comment:
Please don't create a new HC test to validate this so keep infra under control.
Let's just include this new validation as part of any new HC expectation, just like we do for EnsureNoCrashingPods or for the metrics

Reply:
Good, one question, makes sense to include it in to the `t.Cleanup` testing function? I think makes more sense that add it in all the tests that currently creates HC. WDYT?

Reply:
>I think makes more sense that add it in all the tests that currently creates HC

yeh thats' what I meant